mithril-stm/src/key_registration.rs

```diff
@@ -126,23 +127,24 @@ mod tests {
                 }
 
                 let reg = kr.register(stake, pk);
+
                 match reg {
                     Ok(_) => {
                         assert!(keys.insert(pk.vk, stake).is_none());
                     },
-                    Err(RegisterError::KeyRegistered(pk1)) => {
-                        assert!(pk1.as_ref() == &pk.vk);
-                        assert!(keys.contains_key(&pk.vk));
-                    }
-                    Err(RegisterError::KeyInvalid(a)) => {
-                        assert_eq!(fake_it, 0);
-                        assert!(a.verify_proof_of_possession().is_err());
+                    Err(error) =>  match error.downcast_ref::<RegisterError>(){
+                        Some(RegisterError::KeyRegistered(pk1)) => {
+                            assert!(pk1.as_ref() == &pk.vk);
+                            assert!(keys.contains_key(&pk.vk));
+                        },
+                        Some(RegisterError::KeyInvalid(a)) => {
+                            assert_eq!(fake_it, 0);
+                            assert!(a.verify_proof_of_possession().is_err());
+                        },
+                        _ => {println!("Unexpected error: {error}")}
```

Comment:
We probably want to panic here:
```suggestion
                        _ => {panic!("Unexpected error: {error}")}
```